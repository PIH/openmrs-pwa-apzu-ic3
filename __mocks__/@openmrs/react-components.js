import types from 'redux-types';
// mock methods

const visitRest = {

  createVisit: (params) => {

    const visit = params.visit;

    if (visit.patient != 'patient_uuid' || visit.visitType != 'visit_type_uuid' || visit.location != 'location_uuid') {
      throw {
        name: 'Mock Exception',
        message: 'Unable to Submit'
      };
    }
    else {
      return;
    }
  }

};

const encounterRest = {

  createEncounter: (params) => {

    if (params.encounter.encounterType === 'invalid_encounter_type') {
      throw {
        name: 'Mock Exception',
        message: 'Unable to Submit'
      };
    }
    else {
      return;
    }
  }

};

const BASIC_TYPES = [
  'FETCH_SUCCEEDED',
  'FETCH_PENDING',
  'SET_SUCCEEDED',
  'SET_REQUESTED',
  'SET_FAILED',
  'FETCH_FAILED',
  'FETCH_REQUESTED',
  'SUCCEEDED',
  'PENDING',
  'FAILED',
  'REQUESTED'
];

const LOGIN_TYPES = {
  LOGIN: types('login', BASIC_TYPES),
  LOGIN_LOCATIONS: types('login_locations', BASIC_TYPES),
  LOGOUT: types('logout', BASIC_TYPES)
}

const SESSION_TYPES = types('session', BASIC_TYPES)


export {
  visitRest,
  encounterRest,
  LOGIN_TYPES,
  SESSION_TYPES
};
