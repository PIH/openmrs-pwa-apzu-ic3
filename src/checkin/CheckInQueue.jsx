import React from 'react';
//import DatePicker from 'react-16-bootstrap-date-picker';
//import { Button, ButtonToolbar, Grid, Row, Col,FormGroup, ControlLabel } from 'react-bootstrap';
import {
  patientActions,
  CardList,
  patientObjByEncounterTypeFilter,
  selectors,
  PatientCard,
} from '@openmrs/react-components';

import { push } from "connected-react-router";
import { connect } from "react-redux";
import { ENCOUNTER_TYPES, LOCATION_TYPES } from '../constants';
import utils from "../utils";
import ic3PatientActions from '../patient/patientActions';
import IdentifierFilters from '../screening/IdentifierFilters'
import screeningActions from '../screening/actions/actions'

class CheckInQueue extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      appointmentDate: new Date().toISOString()
    };

  }


  handleAppointmentDateChange(value, formattedValue) {
    this.setState(() => ({
      appointmentDate: value,
      formattedAppointmentDate: formattedValue
    }));
  }


  getAppointmentReport(){
    this.props.dispatch(patientActions.clearPatientStore());
    this.props.dispatch(ic3PatientActions.getIC3Patients(this.props.location, utils.formatReportRestDate(this.state.appointmentDate), true));  // getExpectedAppoints == true

  };

  onMountOtherActionCreators() {
    this.props.dispatch(patientActions.clearSelectedPatient());
  }

  redirectToScreeningTaskList() {
    return push('/screening');
  }

  render() {
    const fetchListActionCreator = this.props.fetchListActionCreator ? this.props.fetchListActionCreator :
      () => {
        if (!this.props.updating) {
          this.props.dispatch(ic3PatientActions.getIC3Patients(
            this.props.session.sessionLocation ? this.props.session.sessionLocation.uuid : null, utils.formatReportRestDate(new Date())
          ));
        }
      };

    return (
      <div>
        {/* uncomment this to get an appt search widget that can be used for testing <br />
        <Grid>
          <Row>
            <FormGroup controlId="formApptDate">
              <Col
                componentClass={ControlLabel}
                sm={2}
              >Appointment Date</Col>
              <Col sm={2}>
                <DatePicker
                  id="appt-datepicker"
                  onChange={this.handleAppointmentDateChange.bind(this)}
                  value={this.state.appointmentDate}
                />
              </Col>
              <Col sm={2} >
                <ButtonToolbar>
                  <Button
                    bsSize="small"
                    bsStyle="success"
                    onClick={this.getAppointmentReport.bind(this)}
                    type="submit"
                  >
                    Get Appointment Report
                  </Button>
                </ButtonToolbar>
              </Col>
            </FormGroup>
          </Row>
        </Grid>

        <br />
*/}
        <CardList
          AdditionalSearchFilters={IdentifierFilters}
          additionalSearchFilterFields={[(patient) => patient.identifiers ? patient.identifiers.map(i => i.identifier) : null]}
          card={PatientCard}
          delayInterval={0}
          dispatch={this.props.dispatch}
          fetchListActionCreator={fetchListActionCreator}
          filters={[patientObjByEncounterTypeFilter(ENCOUNTER_TYPES.CheckInEncounterType.uuid, 'exclude')]}
          getPatientIdentifiers={utils.getPatientIdentifiers}
          loading={this.props.updating}
          noDataMessage="No patients to display"
          onMountOtherActionCreators={[
            () => this.props.dispatch(patientActions.clearSelectedPatient())
          ] }
          rowData={ Object.values(this.props.patients) }
          rowSelectedActionCreators={[() => screeningActions.setLastScreeningQueue(this.props.screeningLocation), patientActions.setSelectedPatient, this.redirectToScreeningTaskList.bind(this)]}
          searchFilterFields={['name.fullName', 'name.reverseFullName']}
          sortFields={['name.givenName', 'name.familyName']}
          title="Check-In Queue"
        />

      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    location: state.openmrs.session.sessionLocation ? state.openmrs.session.sessionLocation.uuid : LOCATION_TYPES.UnknownLocation,
    patients: selectors.getPatientStore(state),
    updating: selectors.isPatientStoreUpdating(state),
    session: state.openmrs.session,
    screeningLocation: state.router.location.pathname
  };
};

export default connect(mapStateToProps)(CheckInQueue);
