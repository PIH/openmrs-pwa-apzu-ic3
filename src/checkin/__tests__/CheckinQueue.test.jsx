import React from 'react';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { DataGrid } from '@openmrs/react-components';
import CheckInQueue from '../CheckInQueue';

let props, store;
let mountedComponent;

const mockStore = configureMockStore();

const checkInQueue = () => {
  if (!mountedComponent) {
    mountedComponent = mount(
      <Provider store={store}>
        <CheckInQueue {...props} >
        </CheckInQueue>
      </Provider>);
  }
  return mountedComponent;
};

describe('Component: CheckInQueue', () => {
  beforeEach(() => {
    props = {};
    store = mockStore(
      {
        dispatch: {},
        openmrs: {
          session: {
            sessionLocation: {
              uuid: 'abc'
            }
          },
          patients: {
            set: [],
            selected: null
          }
        },
      });
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    //aexpect(toJson(checkInQueue())).toMatchSnapshot();
    expect(checkInQueue().find(DataGrid).props().rowSelectedActionCreators.length).toBe(2);
    //expect(checkInQueue().find(DataGrid).props().rowSelectedActionCreators[0].name).toBe("redirectToCheckinPageActionCreator");
    //expect(checkInQueue().find(DataGrid).props().rowSelectedActionCreators[0]().payload.args[0]).toBe("/checkin/checkInPage");
  });

});

