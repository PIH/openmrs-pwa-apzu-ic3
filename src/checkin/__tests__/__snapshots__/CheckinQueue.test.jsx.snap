// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CheckInQueue renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CheckInQueue)>
    <CheckInQueue
      dispatch={[Function]}
      representation="custom:(uuid,id,display,identifiers:(uuid,identifier,identifierType:(uuid),preferred),person:(uuid,display,gender,age,birthdate,birthdateEstimated,dead,deathDate,causeOfDeath,names,preferredName,addresses,attributes))"
      rowData={Array []}
    >
      <div>
        <Connect(DataGrid)
          columnDefs={
            Array [
              Object {
                "field": "uuid",
                "headerName": "uuid",
                "hide": true,
              },
              Object {
                "headerName": "Id",
                "valueGetter": "data.identifiers[0].identifier",
              },
              Object {
                "field": "name.givenName",
                "headerName": "Given Name",
              },
              Object {
                "field": "name.familyName",
                "headerName": "Family Name",
              },
              Object {
                "field": "gender",
                "headerName": "Gender",
              },
              Object {
                "field": "age",
                "headerName": "Age",
              },
              Object {
                "headerName": "Checked-in Time",
                "valueGetter": [Function],
              },
            ]
          }
          rowData={Array []}
          rowSelectedActionCreators={
            Array [
              [Function],
            ]
          }
        >
          <DataGrid
            columnDefs={
              Array [
                Object {
                  "field": "uuid",
                  "headerName": "uuid",
                  "hide": true,
                },
                Object {
                  "headerName": "Id",
                  "valueGetter": "data.identifiers[0].identifier",
                },
                Object {
                  "field": "name.givenName",
                  "headerName": "Given Name",
                },
                Object {
                  "field": "name.familyName",
                  "headerName": "Family Name",
                },
                Object {
                  "field": "gender",
                  "headerName": "Gender",
                },
                Object {
                  "field": "age",
                  "headerName": "Age",
                },
                Object {
                  "headerName": "Checked-in Time",
                  "valueGetter": [Function],
                },
              ]
            }
            dispatch={[Function]}
            rowData={Array []}
            rowSelectedActionCreators={
              Array [
                [Function],
              ]
            }
          >
            <div
              style={
                Object {
                  "height": "600px",
                  "width": "100%",
                }
              }
            >
              <label>
                Search:
                <input
                  onChange={[Function]}
                  type="text"
                />
              </label>
              <AgGridReact
                columnDefs={
                  Array [
                    Object {
                      "field": "uuid",
                      "headerName": "uuid",
                      "hide": true,
                    },
                    Object {
                      "headerName": "Id",
                      "valueGetter": "data.identifiers[0].identifier",
                    },
                    Object {
                      "field": "name.givenName",
                      "headerName": "Given Name",
                    },
                    Object {
                      "field": "name.familyName",
                      "headerName": "Family Name",
                    },
                    Object {
                      "field": "gender",
                      "headerName": "Gender",
                    },
                    Object {
                      "field": "age",
                      "headerName": "Age",
                    },
                    Object {
                      "headerName": "Checked-in Time",
                      "valueGetter": [Function],
                    },
                  ]
                }
                enableSorting={true}
                id="omrsGrid"
                onGridReady={[Function]}
                onSelectionChanged={[Function]}
                rowClassRules="rowClassRules"
                rowData={Array []}
                rowSelection="single"
              >
                <div
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                />
              </AgGridReact>
            </div>
          </DataGrid>
        </Connect(DataGrid)>
      </div>
    </CheckInQueue>
  </Connect(CheckInQueue)>
</Provider>
`;
